src/internal.rs

```diff
@@ -203,12 +204,37 @@ pub struct Rav1dContext_intra_edge {
     pub tip_sb64: [EdgeTip; 64],
 }
 
+pub(crate) struct Rav1dContextTaskThread {
+    /// Thread join handle, if this task is on a worker thread. The main thread
+    /// task does not contain a handle.
+    pub(crate) handle: Option<pthread_t>,
+    /// Data shared between the main thread and a worker thread.
+    pub(crate) thread_data: Arc<Rav1dTaskContext_task_thread>,
+}
+
+impl Rav1dContextTaskThread {
+    pub(crate) fn new(
+        handle: Option<pthread_t>,
+        thread_data: Arc<Rav1dTaskContext_task_thread>,
+    ) -> Self {
+        Self {
+            handle,
+            thread_data,
+        }
+    }
+
+    pub(crate) fn flushed(&self) -> bool {
+        self.thread_data.flushed.load(Ordering::Relaxed)
+    }
+}
```

Comment:
I think you can make all the methods and fields `pub` as long as the type itself is `pub(crate)`.